cmd/zed/manage/lakemanage/lakemanage.go

```diff
@@ -0,0 +1,203 @@
+package lakemanage
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/brimdata/zed/api"
+	"github.com/brimdata/zed/api/client"
+	lakeapi "github.com/brimdata/zed/lake/api"
+	"github.com/segmentio/ksuid"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+func Update(ctx context.Context, lk lakeapi.Interface, conf Config, logger *zap.Logger) error {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	if err := conf.Index.getRules(ctx, lk); err != nil {
+		return err
+	}
+	branches, err := getBranches(ctx, conf, lk, logger)
+	if err != nil {
+		return err
+	}
+	group, ctx := errgroup.WithContext(ctx)
+	for _, branch := range branches {
+		branch := branch
+		group.Go(func() error {
+			for _, task := range branch.tasks {
+				if _, err := task(ctx); err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+	}
+	return group.Wait()
+}
+
+func Monitor(ctx context.Context, conn *client.Connection, conf Config, logger *zap.Logger) error {
+	if logger == nil {
+		logger = zap.NewNop()
+	}
+	for {
+		switch err := runMonitor(ctx, conf, conn, logger); {
+		case errors.Is(err, syscall.ECONNREFUSED):
+			logger.Info("cannot connect to lake, retrying in 5 seconds")
+		case err != nil:
+			return err
+		}
+		select {
+		case <-time.After(5 * time.Second):
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+func runMonitor(ctx context.Context, conf Config, conn *client.Connection, logger *zap.Logger) error {
+	lk := lakeapi.NewRemoteLake(conn)
+	if err := conf.Index.getRules(ctx, lk); err != nil {
+		return err
+	}
+	branches, err := getBranches(ctx, conf, lk, logger)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	monitors := make(map[ksuid.KSUID]*monitor)
+	for _, pool := range branches {
+		monitorBranch(ctx, pool, monitors)
+	}
+	return listen(ctx, monitors, conf, conn, logger)
+}
+
+func getBranches(ctx context.Context, conf Config, lk lakeapi.Interface, logger *zap.Logger) ([]*branch, error) {
+	pools, err := lakeapi.GetPools(ctx, lk)
+	if err != nil {
+		return nil, err
+	}
+	var branches []*branch
+	for _, pool := range pools {
+		// XXX For now only track main but this should be configurable and allow
+		// non-main branches as well as monitoring multiple branches.
+		branches = append(branches, newBranch("main", pool, lk, conf, logger))
+	}
+	return branches, nil
+}
+
+func listen(ctx context.Context, monitors map[ksuid.KSUID]*monitor, conf Config, conn *client.Connection, logger *zap.Logger) error {
+	ev, err := conn.SubscribeEvents(ctx)
+	if err != nil {
+		return err
+	}
+	defer ev.Close()
+	lk := lakeapi.NewRemoteLake(conn)
+	for {
+		kind, detail, err := ev.Recv()
+		if err != nil {
+			if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
+				// Ignore EOF error from lost connection.
+				return nil
+			}
+			return err
+		}
+		switch kind {
+		case "pool-new":
+			ev := detail.(*api.EventPool)
```

Comment:
Nits: Why switch on kind instead of the type? And it's surprising to shadow ev with something of a different type.
```suggestion
		switch detail := detail.(type) {
		case *api.EventPool:
```

Reply:
"pool-new" and "pool-delete" both use `*api.EventPool` but mean different things.

Reply:
> I'm not quite sure I understand this. The design as is on this endpoint isn't ideal- i'd like to at some point make these zson, but kept this to avoid a migration for now.

ev is a *client.EventsClient outside this switch statement but a *api.EventWhatever inside it. Maybe do this?

```suggestion
		switch kind {
		case "pool-new":
			detail := detail.(*api.EventPool)
```

Reply:
I can dig it